src/User/UserManager.php

```diff
@@ -198,16 +198,38 @@ public function addRecord($name, $provider_user_id, $token, $additional_data = N
     ];
 
     $user_info = $this->entityTypeManager->getStorage('social_post')->create($values);
-    $user_info->setToken($token);
 
-    // Saves the entity.
-    $user_info->save();
+    if (!$user_info) {
+      return FALSE;
+    }
+
+    try {
+      $user_info->setToken($token);
+
+      // Saves the entity.
+      $user_info->save();
 
-    if ($user_info) {
       return TRUE;
+
+    }
+    catch (\Exception $ex) {
```

Comment:
Try catching for the specific exception `EntityStorageException` instead of the global one.